src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -233,4 +247,34 @@ public virtual NpgsqlOptionsExtension WithRemoteCertificateValidationCallback([C
 
         #endregion Authentication
     }
+
+    public readonly struct RangeMappingInfo
+    {
+        /// <summary>The name of the PostgreSQL range type to be mapped.</summary>
+        public readonly string RangeName;
+        /// <summary>
+        /// The CLR type of the range's subtype (or element).
+        /// The actual mapped type will be an <see cref="NpgsqlRange{T}"/> over this type.
+        /// </summary>
+        public readonly Type SubtypeClrType;
+        /// <summary>
+        /// Optionally, the name of the range's PostgreSQL subtype (or element).
+        /// This is usually not needed - the subtype will be inferred based on <see cref="SubtypeClrType"/>.
+        /// </summary>
+        public readonly string SubtypeName;
```

Comment:
Even if it's a `readonly struct`, I think it would be better to have public properties instead of fields.

Reply:
Just my two cents, but for `readonly struct` I would actually stick with the `public readonly` fields. Adding back the layer of indirection feels....indirect.

Reply:
Agreed, but the common practice is to continue to use properties. Therefore, these fields could be replaced by `readonly` properties. No additional code for back storage is required:

```csharp
public string RangeName { get; }
```

There is an article about `readonly struct`s, immutability and performance. [The ‘in’-modifier and the readonly structs in C#](https://blogs.msdn.microsoft.com/seteplia/2018/03/07/the-in-modifier-and-the-readonly-structs-in-c/).

Anyway I don't think that it's a blocking issue. Just would like to hear @roji's thought on it (:

Reply:
> The compiler checks that the struct is indeed immutable and consists only of readonly fields and/or readonly properties (properties like `public int Foo {get; private set;}` are not readonly).

Ahh, I read that a while back but overlooked the comments on read-only instance properties being eligible! Thanks for pointing that out!